components/function-controller/pkg/controller/function/function_controller.go

```diff
@@ -211,7 +207,8 @@ func (r *ReconcileFunction) Reconcile(req reconcile.Request) (reconcile.Result,
 	buildName := fmt.Sprintf("%s-%s", fn.Name, fnSha[:buildNameSuffixLen])
 	log.Info("Build info", "namespace", fn.Namespace, "name", fn.Name, "buildName", buildName, "imageName", imgName)
 
-	err = r.buildFunctionImage(rnInfo, fn, imgName, buildName)
+	// Run Function build (Tekton TaskRun)
+	fnTr, err := r.buildFunctionImage(rnInfo, fn, imgName, buildName)
```

Comment:
This function now returns the `TaskRun` it has found or created, so we can pass it later on to `setFunctionCondition()` and compute the function status. This way we save ourselves extra (and expensive) calls to `client.Get()`.